apis/v1alpha1/httproute_types.go

```diff
@@ -207,32 +222,67 @@ type HTTPRouteMatch struct {
 	ExtensionRef *RouteMatchExtensionObjectReference `json:"extensionRef" protobuf:"bytes,5,opt,name=extensionRef"`
 }
 
-// HTTPRouteFilter defines a filter-like action to be applied to
-// requests.
-type HTTPRouteFilter struct {
-	// Headers related filters.
-	//
-	// Support: extended
+const (
+	// FilterHTTPReqeustHeader can be used to add or remove an HTTP
+	// header from an HTTP request before it is sent to the upstream target.
+	// Support: core
 	// +optional
-	Headers *HTTPHeaderFilter `json:"headers" protobuf:"bytes,1,opt,name=headers"`
+	FilterHTTPReqeustHeader = "RequestHeader"
+	// TODO(hbagdi): add FilterHTTPRequestMirroring filter
+)
+
+// HTTPRouteFilter defines additional processing steps that must be completed
+// during the request or response lifecycle.
+// HTTPRouteFilters are meant as an extension point to express additional
+// processing that must be done in Gateway implementations. Some examples include
+// request or response modification, implementing authentication strategies,
+// rate-limiting, and traffic shaping.
+// API guarantee/conformance is defined based on the type of the filter.
+// TODO(hbagdi): re-render CRDs once controller-tools supports union tags:
+// - https://github.com/kubernetes-sigs/controller-tools/pull/298
+// - https://github.com/kubernetes-sigs/controller-tools/issues/461
+// +union
+type HTTPRouteFilter struct {
+	// Type identifies the filter to execute.
+	// Types are classified into two conformance-levels:
+	// - Core and extended: These filter types and their corresponding configuration
+	//   is defined in this package. All implementations must implement
+	//   the core filters. Implementers are encouraged to support extended filters.
+	//   Definitions for filter-specific configuration for these
+	//   filters is defined in this package.
+	// - Custom: These filters are defined and supported by specific vendors.
```

Comment:
I may be misunderstanding this, but it seems like there are actually 3 conformance levels described here: core, extended, and custom.

Reply:
That's correct. What's the concern/question here?

Reply:
Line 247: `Types are classified into two conformance-levels`

Reply:
Ah, I'm blind. Updated.